src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function compareDefault(a, b) {
+    if (String(a) > String(b)) {
+      return 1;
+    }
+
+    if (String(a) < String(b)) {
+      return -1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = compareDefault) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction(this[i], this[j]) > 0) {
+          const temp = this[i];
+
+          this[i] = this[j];
+          this[j] = temp;
+        }
```

Comment:
The swapping of elements is done correctly using a temporary variable. As an alternative, you could also use array destructuring to swap values: '[this[i], this[j]] = [this[j], this[i]]'. This can make the code more concise and easier to read.